src/openqaoa-core/problems/vehiclerouting.py

```diff
@@ -0,0 +1,380 @@
+        avoided introducing the constraint subtours=[[4,5,8]]. To additional information
+        about subtours refer to https://de.wikipedia.org/wiki/Datei:TSP_short_cycles.png
+    all_subtours:
+    penalty: float
+        Penalty for the constraints. If the method is 'unbalanced' three values are needed,
+        one for the equality constraints and two for the inequality constraints.
+    method: str
+        Two available methods for the inequality constraints ["slack", "unbalanced"]
+        For 'unblanced' see https://arxiv.org/abs/2211.13914
+    Returns
+    -------
+        An instance of the VRP problem.
+    """
+
+    __name__ = "vehicle_routing"
+
+    def __init__(
+        self,
+        G,
+        pos,
+        n_vehicles,
+        depot: int = 0,
+        subtours: list = [],
+        method: str = "slack",
+        penalty: Union[int, float, list] = 4,
+    ):
+        self.G = G
+        if len(G.nodes) != len(pos):
+            raise ValueError(
+                f"The number of nodes in G is {len(G.nodes)} while the x, y coordinates in pos is {len(pos)}"
+            )
+        self.pos = pos
+        self.n_vehicles = n_vehicles
+        self.depot = depot
+        self.subtours = subtours
+        self.method = method
+        if method == "unbalanced" and len(penalty) != 3:
+            raise ValueError(
+                "The penalty must have 3 parameters [lambda_0, lambda_1, lambda_2]"
+            )
+        else:
+            self.penalty = penalty
+
+    @property
+    def G(self):
+        return self._G
+
+    @G.setter
+    def G(self, input_networkx_graph):
+        if not isinstance(input_networkx_graph, nx.Graph):
+            raise TypeError("Input problem graph must be a networkx Graph.")
+
+        # Relabel nodes to integers starting from 0
+        mapping = dict(
+            zip(input_networkx_graph, range(input_networkx_graph.number_of_nodes()))
+        )
+        self._G = nx.relabel_nodes(input_networkx_graph, mapping)
+
+    @staticmethod
+    def random_instance(**kwargs):
+        """
+        Creates a random instance of the vehicle routing problem, whose graph is
+        random.
+
+        Parameters
+        ----------
+        **kwargs:
+            Required keyword arguments are:
+            n_nodes: int
+                The number of nodes (vertices) in the graph.
+            n_vehicles: int
+                Number of vehicles used in the problem
+            method: str
+                method for the inequality constraints ['slack', 'unbalanced'].
+                For the unbalaced method refer to https://arxiv.org/abs/2211.13914
+                For the slack method: https://en.wikipedia.org/wiki/Slack_variable
+            subtours: list[list]
+                Manually add the subtours to be avoided
+            seed: int
+                Seed for the random problems.
+
+        Returns
+        -------
+            A random instance of the maximal independent set problem.
```

Comment:
Typo in the text, it refers to MIS instead of VSP

Reply:
done!